File: source-connector/src/main/java/sourceconnector/service/producer/BatchProducer.java
package sourceconnector.service.producer;

import offsetmanager.domain.offset.OffsetRecord;
import sourceconnector.domain.batch.MessageBatch;


public interface BatchProducer<T>  extends AutoCloseable {
  void sendBatch(OffsetRecord offsetRecord, MessageBatch<T> messageBatch);
}

File: source-connector/src/main/kotlin/sourceconnector/service/producer/BatchProducer.kt
package sourceconnector.service.producer

import offsetmanager.domain.offset.OffsetRecord
import sourceconnector.domain.batch.MessageBatch
import java.lang.AutoCloseable

interface BatchProducer<T> : AutoCloseable {
  fun sendBatch(
    offsetRecord: OffsetRecord,
    messageBatch: MessageBatch<T>
  ): BatchResult
}
